File: leetcode/1002.find-common-characters/cpp/commonchars_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "commonchars.cpp"

struct testCase
{
    std::vector<std::string> words;
    std::vector<std::string> want;
};

TEST_CASE("CommonChars")
{
    std::vector<testCase> tcs{
        {{"bella", "label", "roller"}, {"e", "l", "l"}},
        {{"cool", "lock", "cook"}, {"c", "o"}},
    };

    Solution s;

    for (auto tc : tcs)
    {
        auto got = s.commonChars(tc.words);
        CHECK(got == tc.want);
    }
}

File: leetcode/1002.find-common-characters/ts/commonchars.test.ts
import { test, expect } from 'vitest'
import commonChars from './commonchars.js'

interface testcase {
    words: string[]
    want: string[]
}

test('commonchars', () => {
    const tcs: testcase[] = [
        { words: ["bella", "label", "roller"], want: ["e", "l", "l"] },
        { words: ["cool", "lock", "cook"], want: ["c", "o"] },
    ]

    for (const tc of tcs) {
        expect(tc.want === commonChars(tc.words))
    }
})
